server/backends/blobstore/util/util.go

```diff
@@ -71,8 +71,47 @@ func Compress(in []byte) ([]byte, error) {
 	return io.ReadAll(&buf)
 }
 
-var BlobPath = func(blobName string) string {
-	return filepath.Join(*pathPrefix, blobName)
+// prefixBlobstore implements interfaces.Blobstore, is a wrapper around
+// an existing blobstore that prefixes all blob names with `storage.path_prefix`.
+type prefixBlobstore struct {
+	blobstore interfaces.Blobstore
+	prefix    string
+}
+
+func NewDefaultPrefixBlobstore(b interfaces.Blobstore) *prefixBlobstore {
+	return NewPrefixBlobstore(b, *pathPrefix)
+}
+
+// NewPrefixBlobstore returns a new prefixBlobstore that wraps the given
+// blobstore and prefixes all blob names with the given prefix.
+// Intent to support testing only, you probably want to use
+// NewDefaultPrefixBlobstore instead.
+func NewPrefixBlobstore(b interfaces.Blobstore, prefix string) *prefixBlobstore {
+	return &prefixBlobstore{b, prefix}
+}
+
+func (p *prefixBlobstore) blobPath(blobName string) string {
+	return filepath.Join(p.prefix, blobName)
+}
```

Comment:
since this func exists I think we can delete `util.BlobPath` now? Do you want to do that in this PR?

Reply:
I did remove util.BlobPath in this PR 🤔 , do you mean removing `prefixBlobstore.blobPath()` in favor of using `filepath.Join()` directly?

Reply:
woops, ignore me - I missed the diff :sweat_smile: